### ISSUE TITLE
Specfile with invalid examples considered valid

### ISSUE THREAD

**Issue by lovebug356** [asyncapi/cli#925]
A specfile with invalid examples is considered a valid specfile when validated with `asyncapi validate`.

This issue can be reproduced with following example. This file contains an invalid example (missing required field):
```
asyncapi: 3.0.0
info:
  title: Account Service
  version: 1.0.0
  description: This service is in charge of processing user signups
channels:
  user/signedup:
    address: user/signedup
    messages:
      subscribe.message:
        $ref: '#/components/messages/UserSignedUp'
operations:
  user/signedup.subscribe:
    action: send
    channel:
      $ref: '#/channels/user~1signedup'
    messages:
      - $ref: '#/components/messages/UserSignedUp'
components:
  messages:
    UserSignedUp:
      examples:
        - name: Invalid Example
          summary: Example is missing required field email
          payload:
            displayName: My Name
      payload:
        type: object
        required:
          - displayName
          - email
        properties:
          displayName:
            type: string
            description: Name of the user
          email:
            type: string
            format: email
            description: Email of the user
```

**Comment by fmvilas** [comment]
I think it would be interesting to have a flag in the CLI. Something like `asyncapi validate --skip-examples`.

**Comment by lovebug356** [comment]
I'm trying with the latest git version of `cli` and `studio`. According to npm, they are using this version:
@asyncapi/parser@3.0.0-next-major-spec.2

**Comment by PradeepJoshiFreelancer** [comment]
@lovebug356 - Can you please check the YML provided, on validating its giving an error at line #18
18:9  error  asyncapi3-operation-messages-from-referred-channel  Operation message does not belong to the specified channel.  operations.user/signedup.subscribe.messages[0]

✖ 1 problem (1 error, 0 warnings, 0 infos, 0 hints)

If I try to uncomment the refrence to "UserSignedUp" it errors at a later line:
 21:18  error  asyncapi-document-resolved  "messages" property type must be array  components.messages.UserSignedUp

✖ 1 problem (1 error, 0 warnings, 0 infos, 0 hints)

**Comment by lovebug356** [comment]
@PradeepJoshiFreelancer : Updated the example (adapted file generated by `asyncapi new file`).

**Comment by lovebug356** [comment]
> What version of the parser are you using?
> 
> I believed the following Spectral rule takes care of such a validation https://github.com/asyncapi/parser-js/blob/master/src/ruleset/v2/ruleset.ts#L119 🤔
> 
> I'm afk, so I can't check this atm.

I see you point to a spectral rule in v2 ruleset. I don't see something similar in v3 ruleset. Could it be that it is only supported for v2 at the moment?

**Comment by PeteAudinate** [comment]
I think that's right: it worked in v2 but the tools currently aren't validating examples in v3.

**Comment by peter-rr** [comment]
@lovebug356 That's right, this example is correctly validated for v2, but not for v3.

@smoya Tested with:
- `@asyncapi/cli`: v1.7.0
- `@asyncapi/parser`: v3.0.9

**Comment by Souvikns** [comment]
[Studio](https://studio.asyncapi.com/?base64=YXN5bmNhcGk6IDMuMC4wCmluZm86CiAgdGl0bGU6IEFjY291bnQgU2VydmljZQogIHZlcnNpb246IDEuMC4wCiAgZGVzY3JpcHRpb246IFRoaXMgc2VydmljZSBpcyBpbiBjaGFyZ2Ugb2YgcHJvY2Vzc2luZyB1c2VyIHNpZ251cHMKY2hhbm5lbHM6CiAgdXNlclNpZ25lZFVwOgogICAgYWRkcmVzczogdXNlci9zaWduZWR1cAogICAgbWVzc2FnZXM6CiAgICAgIFVzZXJTaWduZWRVcDoKICAgICAgICAkcmVmOiAnIy9jb21wb25lbnRzL21lc3NhZ2VzL1VzZXJTaWduZWRVcCcKb3BlcmF0aW9uczoKICBvblVzZXJTaWduVXA6CiAgICBhY3Rpb246IHJlY2VpdmUKICAgIGNoYW5uZWw6CiAgICAgICRyZWY6ICcjL2NoYW5uZWxzL3VzZXJTaWduZWRVcCcKICAgIG1lc3NhZ2VzOgogICAgICAtICRyZWY6ICcjL2NoYW5uZWxzL3VzZXJTaWduZWRVcC9tZXNzYWdlcy9Vc2VyU2lnbmVkVXAnCmNvbXBvbmVudHM6CiAgbWVzc2FnZXM6CiAgICBVc2VyU2lnbmVkVXA6CiAgICAgIGV4YW1wbGVzOgogICAgICAgIC0gbmFtZTogSW52YWxpZCBFeGFtcGxlCiAgICAgICAgICBzdW1tYXJ5OiBFeGFtcGxlIGlzIG1pc3NpbmcgcmVxdWlyZWQgZmllbGQgZW1haWwKICAgICAgICAgIHBheWxvYWQ6CiAgICAgICAgICAgIGRpc3BsYXlOYW1lOiBNeSBOYW1lCiAgICAgIHBheWxvYWQ6CiAgICAgICAgdHlwZTogb2JqZWN0CiAgICAgICAgcmVxdWlyZWQ6IFtkaXNwbGF5TmFtZSwgZW1haWxdCiAgICAgICAgcHJvcGVydGllczoKICAgICAgICAgIGRpc3BsYXlOYW1lOgogICAgICAgICAgICB0eXBlOiBzdHJpbmcKICAgICAgICAgICAgZGVzY3JpcHRpb246IE5hbWUgb2YgdGhlIHVzZXIKICAgICAgICAgIGVtYWlsOgogICAgICAgICAgICB0eXBlOiBzdHJpbmcKICAgICAgICAgICAgZm9ybWF0OiBlbWFpbAogICAgICAgICAgICBkZXNjcmlwdGlvbjogRW1haWwgb2YgdGhlIHVzZXIKCg%3D%3D) is not throwing any error so looks like this is a valid document. Also according to [v3 schema](https://github.com/asyncapi/spec-json-schemas/blob/bee34d8569a18ee2117b4f2b0529a4e883e73638/schemas/3.0.0-without-%24id.json#L2540) exampels should have either `payload` or `headers`. If it has any one of those it should be valid.

**Comment by smoya** [comment]
> @lovebug356 @PeteAudinate That's right, this example is correctly validated for v2, but not for v3.
> 
> @smoya Tested with:
> 
> * `@asyncapi/cli`: v1.7.0
> * `@asyncapi/parser`: v3.0.9

Yeah, I confirm there is no such rule in place in the Parser for v3 of the spec. 

Can any of the maintainers of this repo move this issue to the parser-js repository? cc @Souvikns @magicmatatjahu @derberg

**Comment by Amzani** [comment]
@smoya I can't
<img width="444" alt="Screenshot 2024-07-05 at 10 19 13" src="https://github.com/asyncapi/cli/assets/554438/ec1cef38-80d8-44d7-9de2-ae463e12d437">
You can try on your side

**Comment by Souvikns** [comment]
I can't either, only transfer issues between CLI, and bundler.
